src/lib/components/collections/req-res-section/RequestResponse.svelte

```diff
@@ -10,12 +10,18 @@
   export let activeTab: Observable<TabDocument>;
   export let collectionsMethods: CollectionsMethods;
   export let environmentVariables;
-  export let loggedUserRoleInWorkspace:WorkspaceRole;
+  export let loggedUserRoleInWorkspace: WorkspaceRole;
+  export let _collectionListViewModel;
 </script>
 
 <div class="d-flex">
   <div class="request-response w-100">
-    <PageHeader {activeTab} {collectionsMethods} {loggedUserRoleInWorkspace} />
+    <PageHeader

```

Comment:
we want `PageHeader` to be a dumb component, not a smart one, so passing in `collectionMethods`, `_collectionListViewModel` and other details make `PageHeader` very restrictive and distribute the functionality amongst different components.

Reply:
Existing implementation, refactored

Reply:
It is fine?

Reply:
ideally `collectionMethods` should not be there as a prop itself to `PageHeader`.

It should be this way
```svelte
<PageHeader
   ...
   onSaveButtonClicked={ ...pass a callback here... }
```

Above will ensure the `PageHeader`'s interactions are nothing but plain events which needs to be handled by parent component in better way rather than passing the logic down the tree.